### PR TITLE
Add plain int4_packing_format support for Float8DynamicActivationInt4WeightConfig

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #3714

Summary:
att, this is so that we can save the model for fp8 + int4, and slice it in vllm, and run with the pre shuffled format in the end, the existing conversion works as well: https://github.com/pytorch/ao/blob/e915c0731ca80d4b69e024187829fb231dd70c1e/torchao/quantization/quantize_/workflows/int4/int4_preshuffled_tensor.py#L185

Test Plan:
pytest test/quantization/quantize_/workflows/int4/test_int4_tensor.py
pytest test/prototype/test_tensor_conversion.py -k test_int4_tensor_conversion

checkpoint: https://huggingface.co/jerryzh168/Qwen3-8B-FP8-INT4

vllm bench throughput --model jerryzh168/Qwen3-8B-FP8-INT4
(added print in the conversion and got the print: "torchao tensor conversion, from fp8-int4 plain to fp8-int4 preshuffled")
```
Throughput: 32.18 requests/s, 37071.32 total tokens/s, 4119.04 output tokens/s
Total num prompt tokens:  1024000
Total num output tokens:  128000
```

vllm bench throughput --model Qwen/Qwen3-8B
```
Throughput: 23.87 requests/s, 27499.79 total tokens/s, 3055.53 output tokens/s
Total num prompt tokens:  1024000
Total num output tokens:  128000
```
Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: test/quantization/quantize_/workflows/int4/test_int4_tensor.py [hunk 1]
```diff
@@ -13,40 +13,62 @@
...
-        input = torch.randn(1, 128, dtype=dtype, device=device)
-        linear = torch.nn.Linear(128, 256, dtype=dtype, device=device)
+        input = torch.randn(*M, K, dtype=dtype, device=device)
+        linear = torch.nn.Linear(K, N, dtype=dtype, device=device)
         original = linear(input)
-        quantize_(linear, self.config)
+        quantize_(linear, config)
         quantized = linear(input)
-        self.assertTrue(compute_error(original, quantized) > 20)
+        self.assertTrue(compute_error(original, quantized) > 18)
```
Comment (jerryzh168) [inline test/quantization/quantize_/workflows/int4/test_int4_tensor.py:66]: seems reasonable since we added fp8-int4 in the test as well

File: torchao/quantization/quantize_/workflows/int4/int4_tensor.py [hunk 2]
```diff
@@ -21,10 +21,15 @@
 
 try:
```
Comment (jerryzh168) [inline torchao/quantization/quantize_/workflows/int4/int4_tensor.py:22]: will follow up to remove try/except
